src/Elastic.Apm/Api/Constraints/RequiredAttribute.cs

```diff
@@ -0,0 +1,15 @@
+// Licensed to Elasticsearch B.V under
+// one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+
+using System;
+
+namespace Elastic.Apm.Api.Constraints
+{
+	/// <summary>
+	/// Specifies that a data field value is required.
+	/// </summary>
+	[AttributeUsage(AttributeTargets.Property)]
+	public sealed class RequiredAttribute : Attribute { }

```

Comment:
Just out of interest - do we already have a test on the other PR that makes sure the usage of this attribute is in-sync with the APM Server schema?

Reply:
We do not because it's a chicken/egg scenario in that we only know what is required in the schema but not on the types. The schema required fields are validated in #984 in the mock APM server when receiving integration test data.